Incorrect "implicitly abstract" function when function returns None and implementation is 'pass'

**Bug Report**

**To Reproduce**

```python
import typing

class A(typing.Protocol):
    def require_method(self) -> None: ...

    def optional_method(self) -> None:
        pass

class B(A):
    def require_method(self) -> None:
        print("B")
    
a = B()
```

**Expected Behavior**

This is valid code. The `optional_method` is optional and is implemented with a valid do-nothing body. It doesn't make sense to `return None` from this function.

**Actual Behavior**

```
% mypy --strict file.py
file.py:13: error: Cannot instantiate abstract class "B" with abstract attribute "optional_method"
file.py:13: note: The following method was marked implicitly abstract because it has an empty function body: "optional_method". If it is not meant to be abstract, explicitly return None.
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 0.981
- Mypy command-line flags: --strict
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.10.6

--- a/mypy/messages.py
+++ b/mypy/messages.py
@@ -1341,15 +1341,16 @@ class MessageBuilder:
             return
         if len(attrs_with_none) == 1:
             note = (
-                "The following method was marked implicitly abstract because it has an empty "
-                "function body: {}. If it is not meant to be abstract, explicitly return None."
+                f"{attrs_with_none[0]} is implicitly abstract because it has an empty function "
+                "body. If it is not meant to be abstract, explicitly `return` or `return None`."
             )
         else:
             note = (
                 "The following methods were marked implicitly abstract because they have empty "
-                "function bodies: {}. If they are not meant to be abstract, explicitly return None."
+                f"function bodies: {format_string_list(attrs_with_none)}. "
+                "If they are not meant to be abstract, explicitly `return` or `return None`."
             )
-        self.note(note.format(format_string_list(attrs_with_none)), context, code=codes.ABSTRACT)
+        self.note(note, context, code=codes.ABSTRACT)
 
     def base_class_definitions_incompatible(
         self, name: str, base1: TypeInfo, base2: TypeInfo, context: Context

--- a/test-data/unit/check-protocols.test
+++ b/test-data/unit/check-protocols.test
@@ -3105,14 +3105,14 @@ class NoneCompatible(Protocol):
 
 class A(NoneCompatible): ...
 A() # E: Cannot instantiate abstract class "A" with abstract attributes "f", "g", "h", "i" and "j" \
-    # N: The following methods were marked implicitly abstract because they have empty function bodies: "f", "g", "h", "i" and "j". If they are not meant to be abstract, explicitly return None.
+    # N: The following methods were marked implicitly abstract because they have empty function bodies: "f", "g", "h", "i" and "j". If they are not meant to be abstract, explicitly `return` or `return None`.
 
 class NoneCompatible2(Protocol):
     def f(self, x: int): ...
 
 class B(NoneCompatible2): ...
 B() # E: Cannot instantiate abstract class "B" with abstract attribute "f" \
-    # N: The following method was marked implicitly abstract because it has an empty function body: "f". If it is not meant to be abstract, explicitly return None.
+    # N: "f" is implicitly abstract because it has an empty function body. If it is not meant to be abstract, explicitly `return` or `return None`.
 
 class NoneCompatible3(Protocol):
     @abstractmethod

